src/flint/types/nmod_poly.pyx

```diff
@@ -355,17 +471,85 @@ cdef class nmod_poly(flint_poly):
     def __rmod__(s, t):
         return divmod(t, s)[1]      # XXX
 
-    def __pow__(nmod_poly self, exp, mod):
+    def __pow__(nmod_poly self, exp, mod=None):
         cdef nmod_poly res
         if mod is not None:
-            raise NotImplementedError("nmod_poly modular exponentiation")
+            return self.powmod(exp, mod)
         if exp < 0:
             raise ValueError("negative exponent")
         res = nmod_poly.__new__(nmod_poly)
         nmod_poly_init_preinv(res.val, (<nmod_poly>self).val.mod.n, (<nmod_poly>self).val.mod.ninv)
         nmod_poly_pow(res.val, self.val, <ulong>exp)
         return res
 
+    def powmod(self, e, modulus, mod_rev_inv=None):
```

Comment:
Maybe `pow_mod` is more consistent as a name here along with `inverse_mod` and `compose_mod`.

Reply:
ive done this